### PR TITLE
fix: [CDS-40398]: Show reconcile dialog when pipeline/template save fails

### PR DESCRIPTION
**Pull request by yc147**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 3 Issues 
* Code issue 3 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10635)

File: src/modules/70-pipeline/components/PipelineStudio/SaveTemplateButton/useSaveAsTemplate.tsx [hunk 1]
```diff
@@ -67,8 +71,12 @@ export function useSaveAsTemplate({ data, type, gitDetails }: SaveAsTemplateProp
         ),
         onSave: async (refreshedYaml: string) => {
           await templateConfigDialogHandler.current?.updateTemplate(refreshedYaml)
-        }
+        },
+        isEdit: false
       })
+    } else {
+      clear()
+      showError(getRBACErrorMessage(error), undefined, 'template.save.template.error')
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/components/PipelineStudio/SaveTemplateButton/useSaveAsTemplate.tsx:79]: Typecast error as RBACError.

File: src/modules/70-pipeline/utils/useSaveTemplate.ts [hunk 2]
```diff
@@ -124,42 +111,33 @@ export function useSaveTemplate(TemplateContextMetadata: TemplateContextMetadata
...
-        if (updatedGitDetails?.isNewBranch) {
-          navigateToLocation(latestTemplate, updatedGitDetails)
-        }
-        return { status: response.status }
-      } else {
-        throw response
-      }
-    } catch (error) {
-      if (!isGitSyncEnabled) {
+    const response = await updateExistingTemplateLabelPromise({
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/utils/useSaveTemplate.ts:114]: This function does not exist anymore, please update swagger and use `updateExistingTemplateVersionPromise` everywhere

File: src/modules/70-pipeline/utils/useSaveTemplate.ts [hunk 3]
```diff
@@ -61,24 +58,14 @@ interface UseSaveTemplateReturnType {
...
 export function useSaveTemplate(TemplateContextMetadata: TemplateContextMetadata): UseSaveTemplateReturnType {
-  const {
-    yamlHandler,
-    setLoading,
-    fetchTemplate,
-    deleteTemplateCache,
-    view,
-    isTemplateStudio = true
-  } = TemplateContextMetadata
-  const { isGitSyncEnabled: isGitSyncEnabledForProject, gitSyncEnabledOnlyForFF } = React.useContext(AppStoreContext)
```
Comment (depanshu-dhiman) [inline src/modules/70-pipeline/utils/useSaveTemplate.ts:80]: These FFs are used in multiple places. For gitsync, there was a different error experience it seems, like setLoading was being called conditionally, showError(getRBACErrorMessage(error)) was being called conditionally.
Is this confirmed that it's not the case anymore?

File: src/modules/72-templates-library/components/TemplateStudio/SaveTemplatePopover/SaveTemplatePopover.tsx [hunk 4]
```diff
@@ -69,14 +65,12 @@ function SaveTemplatePopover(
   const { getString } = useStrings()
   const { templateIdentifier } = useParams<TemplateStudioPathProps & ModulePathParams>()
   const [modalProps, setModalProps] = React.useState<ModalProps>()
-  const [menuOpen, setMenuOpen] = React.useState(false)
-  const { isGitSyncEnabled: isGitSyncEnabledForProject, gitSyncEnabledOnlyForFF } = React.useContext(AppStoreContext)
```
Comment (depanshu-dhiman) [inline src/modules/72-templates-library/components/TemplateStudio/SaveTemplatePopover/SaveTemplatePopover.tsx:73]: Same here. Please make sure gitSync doesn't break as gitSimplification changes will also need to be merged soon.

**Review by depanshu-dhiman — CHANGES_REQUESTED** [review]
Please address the comments

File: src/modules/70-pipeline/utils/useSaveTemplate.ts (hunk 3 above)
Comment (yc147) [reply src/modules/70-pipeline/utils/useSaveTemplate.ts:80]: Yes

File: src/modules/70-pipeline/utils/useSaveTemplate.ts (hunk 2 above)
Comment (yc147) [reply src/modules/70-pipeline/utils/useSaveTemplate.ts:114]: Not required for this changes.

File: src/modules/72-templates-library/components/TemplateStudio/SaveTemplatePopover/SaveTemplatePopover.tsx (hunk 4 above)
Comment (yc147) [reply src/modules/72-templates-library/components/TemplateStudio/SaveTemplatePopover/SaveTemplatePopover.tsx:73]: Yes

File: src/modules/70-pipeline/components/PipelineStudio/SaveTemplateButton/useSaveAsTemplate.tsx (hunk 1 above)
Comment (yc147) [reply src/modules/70-pipeline/components/PipelineStudio/SaveTemplateButton/useSaveAsTemplate.tsx:79]: Doesn't make any difference.

**Review by depanshu-dhiman — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**